blog/2024-09-17-better-ecommerce-typescript/index.mdx

```diff
@@ -0,0 +1,507 @@
+
+
+Here are a few more stats to drive the point home.
+
+
+* Over [50% of eCommerce sales](https://www.doofinder.com/en/blog/ecommerce-site-search-statistics) can be traced back to customers who use the search bar! 
+* It turns out that [34% of online shoppers](https://www.doofinder.com/en/blog/non-product-searches) try to search for non-product content. 
+* [87% of shoppers begin](https://www.salesforce.com/blog/2018/08/digital-shopper-first-retail-report-research.html) their product searches online.
+* 43% of users on retail websites [go directly to the search bar](https://www.forrester.com/report/MustHave+eCommerce+Features/-/E-RES89561).
+* Site searchers are [2-3x more likely to convert.](https://neilpatel.com/blog/site-search-killing-your-conversion/)
+
+A few standouts, but clearly, all drive the conclusion that search is a critical component of any e-commerce product.  
+ 
+A few things are responsible for terrible search experiences, namely...
+
+
+* **Poor or no responses (especially with non-product searches):** Users search for problems sometimes, not just products. 
+* **A lack of multilingual search capabilities:** There is an expectation that site search should find users in whatever language they are comfortable in.
+* **Limited “narrow’ search modalities:** Sometimes, text isn’t enough. Pictures do tell a thousand words, multimodal search today is no longer a luxury.
+
+In this article, we’ll look at how we can make eCommerce search better by solving the pitfalls mentioned above. At the end of this post, we’ll have a demo e-commerce application to act as a basis for what a good search is.
+
+
+
+## Understanding Semantic Search
+
+Leveraging semantic search to power our e-commerce search solves each of the e-commerce search pitfalls we detailed above, but what exactly is semantic search? 
+
+Semantic search, also known as vector search, uses machine learning to grasp the context of data by converting it to numerical vectors. It then uses these vectors to find matches based on conceptual similarity for accurate, relevant search results. 
+
+![GIF](./img/semantic-search-flow.gif)
+
+
+Behind the scenes, it uses approximate nearest neighbors (ANN) algorithms to calculate the approximate nearest neighbors to a query, as opposed to kNN algorithms, which calculate the true nearest neighbors.
+
+![GIF](./img/semantic-search.png)
+
+
+Semantic Search with a Vector database will rid us of the most terrible responses and assure of responses for non-keyword searches. The first problem is sorted. 
+
+
+
+## Building our Web application
```

Comment:
What about 
## How to Build an AI-enabled Search Web Application for E-Commerce